backend/app/models/json_web_token.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class JsonWebToken
+  class << self
+    def encode(user_id, expires_at)
+      payload = { user_id:, exp: expires_at.to_i }
+      JWT.encode(payload, Rails.application.credentials.secret_key_base) # デフォルトではHS256アルゴリズムが使用される
+    end
+
+    def decode(token)
+      decoded = JWT.decode(token, Rails.application.credentials.secret_key_base)[0]
+      ActiveSupport::HashWithIndifferentAccess.new(decoded)
+    rescue JWT::DecodeError => e
+      # トークンの有効切れの際はnilを返す
+      Rails.logger.error e.full_message
```

Comment:
デバッグするための情報として残しておきたいならログに残すのは妥当。

シンプルに有効期限切れのケースしか想定できなかったり、いたずらのリクエストが飛んできたりするときぐらいしか想定できないなら、ログを確認したいケースもなさそうなので、ログに残す必要はないかも。
もしくはメッセージだけ警告で残しておくとか


```suggestion
      Rails.logger.warn "JsonWebToken decode failed: #{e.message}"
```